File: 415-add-strings/415-add-strings.java
class Solution {
    public String addStrings(String num1, String num2) {
        StringBuilder result=new StringBuilder();
        int carry=0, sum=0, i=num1.length()-1, j=num2.length()-1;
        while(i>=0 || j>=0){
            sum=carry;
            if(i>=0) sum+=num1.charAt(i--)-'0';
            if(j>=0) sum+=num2.charAt(j--)-'0';
            result.append(sum%10);
            carry=sum/10;
        }
        if(carry!=0) result.append(carry);
        return result.reverse().toString();
    }
}

File: 415-add-strings/415-add-strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        string result="";
        int carry=0, sum=0, i=num1.size()-1, j=num2.size()-1;
        while(i>=0 || j>=0){
            sum=carry;
            if(i>=0) sum+=num1[i--]-'0';
            if(j>=0) sum+=num2[j--]-'0';
            result+=to_string(sum%10);
            carry=sum/10;
        }
        if(carry!=0) result+=to_string(carry);
        reverse(result.begin(),result.end());
        return result;
    }
};